packages/react-core/src/components/Tabs/Tab.tsx

```diff
@@ -117,27 +122,28 @@ const TabBase: React.FunctionComponent<TabProps> = ({
       className={css(
         styles.tabsItem,
         eventKey === localActiveKey && styles.modifiers.current,
-        handleTabClose && styles.modifiers.action,
-        handleTabClose && (isDisabled || isAriaDisabled) && styles.modifiers.disabled,
+        (handleTabClose || actions) && styles.modifiers.action,
+        (isDisabled || isAriaDisabled) && styles.modifiers.disabled,
         childClassName
       )}
       role="presentation"
     >
       {tooltip ? <Tooltip {...tooltip.props}>{tabButton}</Tooltip> : tabButton}
-      {handleTabClose !== undefined && (
-        <span className={css(styles.tabsItemClose)}>
+      {handleTabClose !== undefined && !actions && (
+        <span className={css(styles.tabsItemAction)}>
           <Button
             variant="plain"
             aria-label={closeButtonAriaLabel || 'Close tab'}
             onClick={(event: any) => handleTabClose(event, eventKey, tabContentRef)}
             isDisabled={isCloseDisabled}
           >
-            <span className={css(styles.tabsItemCloseIcon)}>
+            <span className={css(styles.tabsItemActionIcon)}>
               <TimesIcon />
             </span>
           </Button>
         </span>
       )}
+      {actions && actions}
```

Comment:
I may not understand the logic, but it looks like if `!actions` and `handleTabClose` is defined in some way, we'll render the close button, and actions are appended here at the end. I would _probably_ expect the close action to be at the end of any list of actions, and could still be rendered if a user has passed their own actions - meaning we remove `!actions` from the close button logic, and move this `{actions}` output before the close button. wdyt?

@mcarrano @mceledonia @mmenestr this PR allows users to pass their own actions to a tab, along with the help popover and close button. WDYT about the position of the close button and/or the help button? Should the close button always come after any other actions that the user defines? And should the help button always follow the text or could that go anywhere?

Reply:
The close button from `handleTabClose` and `actions` shouldn't be rendered at the same time (if actions exist then the close button here shouldn't render as it expects actions to pass in a close action). 

`handleTabClose` was the initial way I implemented the close button and is controlled via a single prop on the whole `Tabs` rather than individually so I left it in currently.

Reply:
Thinking on it, I'll move `actions` to render before `handleTabClose` and let both render at once (so a user could define a close action for everything and use `actions` for help buttons). It would look weird if a user passed a close button in `actions` as well as `handleTabClose`, though I'm not sure if this would be a common use case.

Reply:
I think the way you have it now makes sense - Text, then icon, then close. The only thing that looks a bit weird is the spacing. How come in the Help action example, the spacing between the text and the icon is 8px, but in the help and close actions example it's only 4px? I feel like it should be the same - 8px for both!

Reply:
So if I understand correctly this allows for passing any set of actions to a tab? I guess this is OK, but yes, I expect close to always by the last action and  help following the text.

Reply:
@mmenestr the 4px spacer (close and help example) is actually what's coming from core, the 8px spacer in the help example was a mistake since we were missing a class. That reduced spacing is to associate the help popover with the text more than other generic actions and the close action. You can see the spacing outlines in the core issue - https://github.com/patternfly/patternfly/pull/5213#issue-1434013358

FWIW, here's what it looks like now:

![pf4 patternfly org_components_tabs](https://user-images.githubusercontent.com/35148959/202038821-417c89ed-e081-48b2-bead-3efe8aeab50a.png)

And here's with equal spacing.

![pf4 patternfly org_components_tabs (1)](https://user-images.githubusercontent.com/35148959/202038935-18516b7c-91c5-48bd-a9d0-872b49a0f589.png)

Think we should make that update?

Reply:
I think having the flexibility of the generic actions prop is useful. The example shows the basic use case of help + close which most will probably look at when implementing, and if a user wants to deviate or add something (like a cog icon for some kind of "settings" pop up for instance) they have the option to do so.